### PR TITLE
update detik to 1.3.2, add pre-commit, add tests, add dependabot, add macos matrix

### PR DESCRIPTION
**Pull request by brokenpip3**

### CODE REVIEW

File: tests/1-example-tests.bats [hunk 1]
```diff
@@ -0,0 +1,72 @@
+setup() {
+    load 'tests_helper'
+    _tests_helper
+}
+
+
+@test "0: Pre: Create file and dir" {
```
Comment (martin-schulze-vireso) [inline tests/1-example-tests.bats:7]: Why doesn't this have a tag?

File: tests/1-example-tests.bats [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  run mkdir testing
+  run touch testing/example
+  assert_success
+}
+
+#bats test_tags=github:true
+@test "1: Testing file existence" {
+  run ls testing/example
```
Comment (martin-schulze-vireso) [inline tests/1-example-tests.bats:15]: Depending on other tests ist bad.

**Review by martin-schulze-vireso — APPROVED** [review]

File: tests/1-example-tests.bats (hunk 1 above)
Comment (brokenpip3) [reply tests/1-example-tests.bats:7]: Thanks fixed!

File: tests/1-example-tests.bats (hunk 2 above)
Comment (brokenpip3) [reply tests/1-example-tests.bats:16]: I reduce the number and make all the tests without dependencies between each other. Thanks!

**Pull request merged**